pallets/phala/src/registry.rs

```diff
@@ -373,6 +381,41 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(0)]
+		pub fn update_worker_endpoint(
+			origin: OriginFor<T>,
+			signed_endpoint: SignedWorkerEndpoint,
+			signature: Vec<u8>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			// Validate the signature
+			ensure!(
+				signature.len() == 64,
+				Error::<T>::InvalidSignatureLength
+			);
+			let sig = sp_core::sr25519::Signature::try_from(signature.as_slice())
+				.or(Err(Error::<T>::MalformedSignature))?;
+			let encoded_data = signed_endpoint.versioned_endpoints.encode();
+
+			ensure!(
+				sp_io::crypto::sr25519_verify(&sig, &encoded_data, &signed_endpoint.pubkey),
+				Error::<T>::InvalidSignature
+			);
+
+			// Validate the time
+			let expiration = 4 * 60 * 60 * 1000; // 4 hours
+			let now = T::UnixTime::now().as_millis().saturated_into::<u64>();
+			ensure!(
+				signed_endpoint.signing_time + expiration <= now,
+				Error::<T>::InvalidEndpointBlockTime
+			);
+
+			Endpoints::<T>::insert(signed_endpoint.pubkey, signed_endpoint.versioned_endpoints);
```

Comment:
We should also check the worker `signed_endpoint.pubkey` exist (the worker is registered). Otherwise a malicious guy can insert as many endpoint as they want.

Reply:
Done